include/kokkos-utils/callbacks/TimerListener.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef KOKKOS_UTILS_CALLBACKS__TIMERLISTENER_HPP
+#define KOKKOS_UTILS_CALLBACKS__TIMERLISTENER_HPP
+
+#include "Kokkos_NumericTraits.hpp"
+
+#include "kokkos-utils/callbacks/Listener.hpp"
+#include "kokkos-utils/callbacks/Matcher.hpp"
+#include "kokkos-utils/concepts/ExecutionSpace.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+// template <typename...>
+// class TimerListener;
+
+template <typename TimerType, typename EventType>
+concept TimerStartFor = Event<EventType> && requires (TimerType& timer, const EventType& event) {
+    { timer.start(event) } -> std::same_as<void>;
+};
+
+template <typename TimerType, typename EventType>
+concept TimerStopFor = Event<EventType> && requires (TimerType& timer, const EventType& event) {
+    { timer.stop(event) } -> std::same_as<void>;
+};
+
+/**
+ * @brief Listener that starts a timer when a begin event is received and stops it
+ *        when a corresponding end event is received.
+ *
+ * The start event is selected by requiring it to be of type @p BeginEventType and
```

Comment:
If the `BeginEndMatcherType` stores the types of the begin and end evens, you can also use them to update this docstring here.

Reply:
I'll update the docstrings once the code is converged (remember that I'm inheriting your old documentation too).